src/mo_frontend/definedness.ml

```diff
@@ -178,12 +178,15 @@ and dec msgs d = match d.it with
   | VarD (i, e) -> (M.empty, S.singleton i.it) +++ exp msgs e
   | TypD (i, tp, t) -> (M.empty, S.empty)
   | ClassD (csp, i, tp, p, t, s, i', dfs) ->
+    let extern = if s.it = Type.Actor then Some i' else None in
     (M.empty, S.singleton i.it) +++ delayify (
-      group msgs (dec_fields msgs dfs @ class_self d.at i') /// pat msgs p /// shared_pat msgs csp
+      group msgs ~extern (dec_fields msgs dfs @ class_self d.at i' s.it) /// pat msgs p /// shared_pat msgs csp
```

Comment:
Instead of adding the ~extern optional parameter, why not just add (class_self ... ) to beginning or end of the decs argument, depending on whether this is an actor (at front) or at other object (at end).

Reply:
(and then rename class_self to obj_self...)

Reply:
Well, that would break
```
$ cat test/run-drun/self-shadow.mo
actor foo {
  public func foo() {};

  flexible func go() : async () {
    let bar = actor bar { public func bar() {} }
  };
}
```
right? See also the discussion in #1104.

Reply:
Ok, I had a go - you are right about the shadowing but that's possible to account for #4736

Reply:
Merged, thanks!